nucypher/cli/commands/stake.py

```diff
@@ -625,7 +625,7 @@ def increase(general_config: GroupGeneralConfig,
             emitter.echo(MAXIMUM_STAKE_REACHED, color='red')
             raise click.Abort
 
-        stake_value_range = click.FloatRange(min=0, max=upper_limit.to_tokens(), clamp=False)
+        stake_value_range = DecimalRange(min=0, max=upper_limit.to_tokens(), clamp=False)
         value = click.prompt(PROMPT_STAKE_INCREASE_VALUE.format(upper_limit=upper_limit),
                              type=stake_value_range)
     value = NU.from_tokens(value)
```

Comment:
Any concern here since `from_tokens(...)` takes a parameter with a type hint of `Union[int, float, str]`, which does not include Decimal?

Stepping back, it seems odd that `NU.to_tokens()` returns a `Decimal` but `NU.from_tokens(...)` takes a `Union[int, float, str]`. Does `Decimal` need to be added to that `Union`? It seems this is what currency.to_wei (which is eventually called within `NU.__init__()`) ended up already doing:
```python
def to_wei(number: Union[int, float, str, decimal.Decimal], unit: str) -> int:
```

Reply:
Yes, you're right. I'm going to add Decimal to those type hints. Actually, now that you mention, we shouldn't encourage using `float` precisely because of the issues this PR is trying to solve.